src/Components/test/testassets/BasicTestApp/FileDownloaderComponent.razor

```diff
@@ -0,0 +1,36 @@
+@using Microsoft.AspNetCore.Components
+
+@inject IFileDownloader FileDownload
+
+<h1>File configuration</h1>
+File name:
+<br />
+
+<input type="text" @bind-value="@fileName" />
+<br />
+
+<h1>File download</h1>
+
+<button @onclick="SaveFile"> Save </button>
+
+@code {
+    private string fileName = "file.txt";
+
+    private Stream GetFileStream()
+    {
+        var data = new byte[100 * 1024];
+        for (var i = 0; i < data.Length; i++)
+        {
+            data[i] = (byte)(i % 256);
+        }
+        var fileStream = new MemoryStream(data);
+
+        return fileStream;
+    }
+
+    private async Task SaveFile()
+    {
+        var data = GetFileStream();
+        await FileDownload.SaveAs(fileName, data);
+    }
+}
```

Comment:
This looks good. Do you know if Selenium gives us any sensible way to automate a test about file downloads? I'm unsure.

Reply:
We could hijack the JS on the Blazor instance to skip the part about actually showing the dialog (I do not believe there is a way to deal with it in Selenium)

Reply:
Sounds a bit complicated. We could add it to CTI tests I guess, even though it's a bit unusual to test a relatively low-level feature via CTI.

Reply:
> Do you know if Selenium gives us any sensible way to automate a test about file downloads?

Unfortunately, this doesn't look like something that's currently supported (in fact it's discouraged):
https://www.selenium.dev/documentation/test_practices/discouraged/file_downloads/

There are workarounds, but I don't think it's worth the additional long term maintenance complexity. We'll have to defer this to CTI as you mention.